src/ImageSharp/Formats/Pbm/BinaryEncoder.cs

```diff
@@ -0,0 +1,203 @@
+// Copyright (c) Six Labors.
+// Licensed under the Apache License, Version 2.0.
+
+using System;
+using System.Buffers;
+using System.IO;
+using SixLabors.ImageSharp.Memory;
+using SixLabors.ImageSharp.PixelFormats;
+
+namespace SixLabors.ImageSharp.Formats.Pbm
+{
+    /// <summary>
+    /// Pixel encoding methods for the PBM binary encoding.
+    /// </summary>
+    internal class BinaryEncoder
+    {
+        /// <summary>
+        /// Decode pixels into the PBM binary encoding.
+        /// </summary>
+        /// <typeparam name="TPixel">The type of input pixel.</typeparam>
+        /// <param name="configuration">The configuration.</param>
+        /// <param name="stream">The bytestream to write to.</param>
+        /// <param name="image">The input image.</param>
+        /// <param name="colorType">The ColorType to use.</param>
+        /// <param name="maxPixelValue">The maximum expected pixel value</param>
+        /// <exception cref="InvalidImageContentException">
+        /// Thrown if an invalid combination of setting is requested.
+        /// </exception>
+        public static void WritePixels<TPixel>(Configuration configuration, Stream stream, ImageFrame<TPixel> image, PbmColorType colorType, int maxPixelValue)
+            where TPixel : unmanaged, IPixel<TPixel>
+        {
+            if (colorType == PbmColorType.Grayscale)
+            {
+                if (maxPixelValue < 256)
+                {
+                    WriteGrayscale(configuration, stream, image);
+                }
+                else
+                {
+                    WriteWideGrayscale(configuration, stream, image);
+                }
+            }
+            else if (colorType == PbmColorType.Rgb)
+            {
+                if (maxPixelValue < 256)
+                {
+                    WriteRgb(configuration, stream, image);
+                }
+                else
+                {
+                    WriteWideRgb(configuration, stream, image);
+                }
+            }
+            else
+            {
+                WriteBlackAndWhite(configuration, stream, image);
+            }
+        }
+
+        private static void WriteGrayscale<TPixel>(Configuration configuration, Stream stream, ImageFrame<TPixel> image)
+            where TPixel : unmanaged, IPixel<TPixel>
+        {
+            int width = image.Size().Width;
+            int height = image.Size().Height;
```

Comment:
```suggestion
            int width = image.Width;
            int height = image.Height;
```

Also on other places.

Reply:
Indeed, I corrected all the Encoders. Decoders where OK.

Reply:
Indeed, I corrected all the Encoders. Decoders where OK.